src/test/java/com/scottlogic/AuthorPostSorterTest.java

```diff
@@ -10,66 +8,109 @@
 
         assertEquals(expected, actual);
     }
 
     @Test
-    void sortCapital() {
-        UserPost userPost1 = new UserPost("Joe Bloggs",
+    void sort_unsorted3Lower_sorted3Lower() {
+        UserPost userPost1 = new UserPost("joe bloggs",
                 OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
                 "Hello World!", 2);
 
-        UserPost userPost2 = new UserPost("Joe Bloggs",
+        UserPost userPost2 = new UserPost("joe bloggs",
                 OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
                 "Another example post.", 1);
 
-        UserPost userPost3 = new UserPost("Jane Smith",
+        UserPost userPost3 = new UserPost("jane smith",
                 OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
                 "An example of a post \nwith lines breaks.", 3);
 
-        List<UserPost> actual = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> inputList = Arrays.asList(userPost1, userPost2, userPost3);
         List<UserPost> expected = Arrays.asList(userPost3, userPost1, userPost2);
 
-        actual = new AuthorPostSorter().sort(actual);
+        List<UserPost> actual = new AuthorPostSorter().sort(inputList, SortOrder.ASC);
 
         assertEquals(expected, actual);
     }
 
     @Test
-    void sortLower() {
-        UserPost userPost1 = new UserPost("joe bloggs",
+    void sort_unsorted3ItemsDESC_sorted3ItemsDESC() {
+        List<UserPost> inputList = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> expected = Arrays.asList(userPost1, userPost2, userPost3);
+
+        List<UserPost> actual = new AuthorPostSorter().sort(inputList, SortOrder.DESC);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void sort_unsorted2Num_sorted2Num() {
+        UserPost userPost1 = new UserPost("2",
                 OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
                 "Hello World!", 2);
 
-        UserPost userPost2 = new UserPost("joe bloggs",
+        UserPost userPost2 = new UserPost("3",
                 OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
                 "Another example post.", 1);
 
-        UserPost userPost3 = new UserPost("jane smith",
+        UserPost userPost3 = new UserPost("1",
                 OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
                 "An example of a post \nwith lines breaks.", 3);
 
-        List<UserPost> actual = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> inputList = Arrays.asList(userPost1, userPost2, userPost3);
         List<UserPost> expected = Arrays.asList(userPost3, userPost1, userPost2);
 
-        actual = new AuthorPostSorter().sort(actual);
+        List<UserPost> actual = new AuthorPostSorter().sort(inputList, SortOrder.ASC);
 
         assertEquals(expected, actual);
     }
```

Comment:
Another edge case that might be worth testing is the mutability of the input, by doing an assert on a copy of the input after the sort has run you can check that it is the same as before. I would advise doing this in a different test and not checking that the expected output is the same as the actual output within the same test to further the goal of making the problem really clear from the test that is failing. For example if this started failing because of a stability issue on the output you might also think a mutation issue is happening

Reply:
Sorry could you try and clarify this? Are you saying that I should write a test that parses a sorted list, and checks that the output has not changed from he input?